docs/tutorial/index.org

```diff
@@ -0,0 +1,1105 @@
+|   | * ...         |  | * ...         |  | * ...         |   |
+|   | * container n |  | * container n |  | * container n |   |
+|   +---------------+  +---------------+  +---------------+   |
+|                                                             |
++-------------------------------------------------------------+
+#+END_EXAMPLE
+
+
+As we can see in this figure, EnOS will be in charge of provisioning
+the /compute/, /control/ and /network/ nodes. In particular, EnOS will
+deploy Docker containers inside each nodes, which correspond to
+OpenStack services. For instance, the /control/ node will host the
+~nova-api~ and ~nova-scheduler~ containers while the /compute/ node
+will host the ~nova-compute~ and ~nova-libvirt~ containers to provide
+VM hypervisor mechanisms.
+
+#+BEGIN_note
+Note that to deploy on G5K, we need a dedicated node to run EnOS
+because it is discouraged to run experiments on the frontend. This
+restriction is meant to avoid disturbing other users that are logged,
+since the frontend node has limited resources. On a regular
+deployment, EnOS could be run directly from your laptop.
+#+END_note
+
+* Set the /enos/ node and install EnOS
+The first step is to determine on which cluster you will deploy
+OpenStack. To that end, you can run ~funk~ (Find yoUr Nodes on g5K)
+from any frontend to see the availability on G5K:
+
+#+BEGIN_SRC sh
+# laptop:~$
+ssh nantes.g5k
+# fnantes:~$
+funk -w 4:00:00
+#+END_SRC
+
+In this example, we check the availability of G5K’s clusters for the
+next four hours (adapt the time regarding your situation). Note that
+you can adapt the time of your reservation afterward, using the
+~oarwalltime~ command[fn:oarwalltime]. Find a cluster with at least
+four nodes available before going further. Once it is done, reach the
+cluster’s site first, and then, get a new machine which we will use as
+our /enos/ node. In this document, we target the parapide cluster,
+located in the Rennes site:
+
+#+BEGIN_SRC sh
+# fnantes:~$
+ssh rennes
+# fnantes:~$ -- Not mandatory, but recommended
+tmux
+# fnantes:~$ -- Let's connect to the enos node
+oarsub -I -l "nodes=1,walltime=4:00:00" -p "cluster='parapide'"
+#+END_SRC
+
+Here, we get a new machine in interactive mode ({{{ie}}}
+~-I~) for the next four hours from the parapide cluster. If it
+succeeds you should be directly connected to this node (check your
+prompt).
+
+#+BEGIN_note
+Note that we created a ~tmux~ session in order to be resilient to any
+network failure during your ssh session. Whenever you want to restore
+this session, you can connect to the frontend and attach to your tmux
+session, as follows:
+
+#+BEGIN_SRC sh
+# laptop:~$
+ssh rennes.g5k
+# frennes:~$ -- Stands for "tmux attach"
+tmux a
+#+END_SRC
+#+END_note
+
+Download EnOS in your working directory, and install it:
+
+#+BEGIN_SRC sh :tangle ../../tests/functionnal/tests/tutorial/enos-node.sh
+# enos:~$
+mkdir -p ~/enos-myxp; cd ~/enos-myxp
```

Comment:
We maybe should not tangle this line, because when executing the `enos-node.sh` the `reservation.yaml` won't be found

Reply:
Actually, it will be found. The listing lst:reservation.yaml is tangle into reservation.yaml.

Reply:
I'm not sure about this, it seems that `reservation.yaml` is tangled to `../../tests/functionnal/tests/tutorial/reservation.yaml`

Reply:
Yes, according to `enos-node.sh`, `enos deploy -f reservation.yam` is run in `../../tests/functionnal/tests/tutorial/enos-myxp` where `reservation.yaml` isn't found.

We could create a new untangled code block that forces the reader of the TP to change directory. In the same time jenkins won't change directory as it reads the tangled output.

Reply:
You are SO right -- the fact it took me 5 comments to figure this out is a manifest marker  of my need of vacation! I will split this block source to put the `mkdir; cd` out of the tangled code.

Reply:
done